includes/class-content-distribution.php

```diff
@@ -0,0 +1,422 @@
+	 */
+	protected static function get_post_payload( $post ) {
+		$post = get_post( $post );
+		if ( ! $post ) {
+			return new WP_Error( 'invalid_post', __( 'Invalid post.', 'newspack-network' ) );
+		}
+
+		$config = self::get_post_config( $post );
+		return [
+			'site_url'  => get_bloginfo( 'url' ),
+			'post_id'   => $post->ID,
+			'config'    => $config,
+			'post_data' => [
+				'title'       => html_entity_decode( get_the_title( $post->ID ), ENT_QUOTES, get_bloginfo( 'charset' ) ),
+				'date'        => $post->post_date,
+				'slug'        => $post->post_name,
+				'post_type'   => $post->post_type,
+				'raw_content' => $post->post_content,
+				'content'     => self::get_processed_post_content( $post ),
+				'excerpt'     => $post->post_excerpt,
+				// @ TODO: Add meta, featured image and taxonomies.
+			],
+		];
+	}
+
+	/**
+	 * Get the processed post content for distribution.
+	 *
+	 * @param WP_Post $post The post object.
+	 *
+	 * @return string The post content.
+	 */
+	protected static function get_processed_post_content( $post ) {
+		global $wp_embed;
+		/**
+		 * Remove autoembed filter so that actual URL will be pushed and not the generated markup.
+		 */
+		remove_filter( 'the_content', [ $wp_embed, 'autoembed' ], 8 );
+		// Filter documented in WordPress core.
+		$post_content = apply_filters( 'the_content', $post->post_content );
+		add_filter( 'the_content', [ $wp_embed, 'autoembed' ], 8 );
+		return $post_content;
+	}
+
+	/**
+	 * Get a linked post given the distributed post hash.
+	 *
+	 * @param string $post_type The post type.
+	 * @param string $post_hash The distributed post hash.
+	 *
+	 * @return WP_Post|null The linked post or null if not found.
+	 */
+	protected static function get_linked_post( $post_type, $post_hash ) {
+		$posts = get_posts(
+			[
+				'post_type'      => $post_type,
+				'post_status'    => [ 'publish', 'pending', 'draft', 'auto-draft', 'future', 'private', 'inherit', 'trash' ],
+				'posts_per_page' => 1,
+				'meta_query'     => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+					[
+						'key'   => self::POST_HASH_META,
+						'value' => $post_hash,
+					],
+				],
+			]
+		);
+		if ( empty( $posts ) ) {
+			return null;
+		}
+		return $posts[0];
+	}
+
+	/**
+	 * Set a post as unlinked.
+	 *
+	 * This will prevent the post from being updated when the distributed post is
+	 * updated.
+	 *
+	 * @param int  $post_id  The post ID.
+	 * @param bool $unlinked Whether to set the post as unlinked. Default is true.
+	 *
+	 * @return void|WP_Error Void on success, WP_Error on failure.
+	 */
+	public static function set_post_unlinked( $post_id, $unlinked = true ) {
+		if ( ! $post_id || ! get_post( $post_id ) ) {
+			return new WP_Error( 'invalid_post', __( 'Invalid post.', 'newspack-network' ) );
+		}
+		update_post_meta( $post_id, self::POST_UNLINKED_META, (bool) $unlinked );
+
+		// If the post is being re-linked, update content.
+		if ( ! $unlinked ) {
```

Comment:
double negative variables are usually pretty confusing to read. "If Unlinked = false it means that yes, it's linked".. Maybe consider having a LINKED state, with straight forward boolean value that will not be confusing

Reply:
I explored alternatives and I'm not sure it improves readability nor maitainability. Ultimately, the post is always _linked_ and the `insert_linked_post()` method is still called when "unlinked".

Having `unlinked` as the post meta also makes it easier to manage the different state, as we'll replace the content once the flag is removed. Having it the other way around adds some complexity that I don't think justifies the change.

Perhaps we can find a different name for `unlinked` that's not a negative so we can preserve the logic?

Reply:
> Perhaps we can find a different name for unlinked that's not a negative so we can preserve the logic?

Yeah, that would be nice, but I can't think of anything good.

Not the end of the world, if you've explored alternatives and think this is more readable let's go with it

Reply:
How about `standalone`?

Reply:
yeah... meh 

What can help as well is if we had helper methods like `set_post_as_linked` ... `is_post_linked`, that would be shortcuts for the `...unlinked( ..., false)` methods